src/operations/generate/NewRepoCreationParameters.ts

```diff
@@ -0,0 +1,59 @@
+
+import { MappedParameter, MappedParameters, Parameter, Secret, Secrets } from "../../decorators";
+import { GitHubNameRegExp } from "../common/params/gitHubPatterns";
+import { RepoId } from "../common/RepoId";
+
+/**
+ * Parameters common to all generators that create new repositories
+ */
+export class NewRepoCreationParameters implements RepoId {
+
+    @Secret(Secrets.userToken(["repo", "user"]))
+    public githubToken;
+
+    @MappedParameter(MappedParameters.GitHubOwner)
+    public targetOwner: string;
+
+    @Parameter({
+        pattern: GitHubNameRegExp.pattern,
+        displayName: "Target Repository Name",
+        description: "name of the target repository",
+        validInput: GitHubNameRegExp.validInput,
+        minLength: 1,
+        maxLength: 50,
+        required: true,
+        order: 1,
+    })
+    public targetRepo: string;
+
+    @Parameter({
+        displayName: "Project Description",
+        description: "short descriptive text describing the new project",
+        pattern: /.*/,
```

Comment:
Should this be anchored? We probably no longer require that, right? If we don't require it, there's no reason to do it here.